src/init/starship.zsh

```diff
@@ -7,6 +7,15 @@
 # after drawing the prompt. This ensures that the timing for one command is only
 # ever drawn once (for the prompt immediately after it is run).
 
+zmodload zsh/parameter  # Needed to access jobstates variable for NUM_JOBS
+
+starship_render() {
+    # Use length of jobstates array as number of jobs. Expansion fails inside
+    # quotes so we set it here and then use the value later on.
+    NUM_JOBS=$#jobstates
```

Comment:
How about getting `NUM_JOBS` (or better `STARSHIP_NUM_JOBS`) in `starship_precmd()` like `STARSHIP_CMD_STATUS` and keeping the current approach of only setting `PROMPT` once?

Reply:
From a couple of tests, I don't think you can set it just once:
<img width="410" alt="Screenshot 2021-04-25 at 19 33 59" src="https://user-images.githubusercontent.com/1850188/116005028-31751e00-a5fd-11eb-9d56-0cc0501534e9.png">

The updated value of `$STARSHIP_NUM_JOBS` is never read again...

Reply:
It is definitely possible to set `PROMPT` once and keep it updated by updating the variable holding `jobstates` length. See the comment below for the patch doing exactly that. :arrow_down: